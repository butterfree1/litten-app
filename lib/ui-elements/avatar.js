/**
 * @format
 * @flow
 */

import { StyleSheet, View } from 'react-native'
import UIImage from 'ui-elements/image'
import {
  USER_AVATAR_SIZE_LARGE,
  USER_AVATAR_SIZE_MEDIUM,
  USER_AVATAR_SIZE_MINI,
} from 'utils/constants'

const UIAvatar: (args: any) => React$Node = ({
  containerStyle,
  size = 'mini',
  source,
  style,
  ...otherProps
}) => {
  const getSize = () => {
    switch (size) {
      case 'large':
        return styles.uiAvatarSizeLarge
      case 'medium':
        return styles.uiAvatarSizeMedium
      default:
        return styles.uiAvatarSizeMini
    }
  }

  return (
    <View style={[containerStyle, styles.uiAvatarImageContainer, getSize()]}>
      {source && (
        <UIImage
          source={source}
          resizeMode="cover"
          style={[getSize(), style]}
          {...otherProps}
        />
      )}
    </View>
  )
}

const styles = StyleSheet.create({
  uiAvatarImageContainer: {
    overflow: 'hidden',
    alignItems: 'center',
    justifyContent: 'center',
  },
  uiAvatarSizeLarge: {
    height: USER_AVATAR_SIZE_LARGE,
    width: USER_AVATAR_SIZE_LARGE,
    borderRadius: USER_AVATAR_SIZE_LARGE,
  },
  uiAvatarSizeMedium: {
    height: USER_AVATAR_SIZE_MEDIUM,
    width: USER_AVATAR_SIZE_MEDIUM,
    borderRadius: USER_AVATAR_SIZE_MEDIUM,
  },
  uiAvatarSizeMini: {
    height: USER_AVATAR_SIZE_MINI,
    width: USER_AVATAR_SIZE_MINI,
    borderRadius: USER_AVATAR_SIZE_MINI,
  },
})

export default UIAvatar
