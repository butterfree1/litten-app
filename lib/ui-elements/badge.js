/**
 * @format
 * @flow
 */

import { StyleSheet, Text, View } from 'react-native'
import {
  UI_BADGE_ELEMENT_OFFSET,
  UI_BADGE_PADDING,
  UI_BADGE_SIZE,
} from 'utils/constants'
import colors from 'styles/colors'

const UIBadge: (args: any) => React$Node = ({
  children,
  number,
  showNonPositive = false,
}) => (
  <View style={styles.uiBadgeContainer}>
    {Number.isInteger(number) &&
      (showNonPositive || (!showNonPositive && number > 0)) && (
        <View style={styles.uiBadgeNumber}>
          <Text style={styles.uiBadgeNumberText}>{number}</Text>
        </View>
      )}
    {children}
  </View>
)

const styles = StyleSheet.create({
  uiBadgeContainer: {
    position: 'relative',
  },
  uiBadgeNumber: {
    position: 'absolute',
    alignItems: 'center',
    justifyContent: 'center',
    minHeight: UI_BADGE_SIZE,
    minWidth: UI_BADGE_SIZE,
    top: -UI_BADGE_ELEMENT_OFFSET / 2,
    right: -UI_BADGE_ELEMENT_OFFSET,
    padding: UI_BADGE_PADDING,
    borderRadius: UI_BADGE_SIZE / 2,
    backgroundColor: colors.red,
    zIndex: 1,
  },
  uiBadgeNumberText: {
    color: colors.white,
    fontSize: 10,
    fontWeight: '600',
    textAlign: 'center',
  },
})

export default UIBadge
