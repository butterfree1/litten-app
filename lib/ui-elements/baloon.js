/**
 * @format
 * @flow
 */

import { StyleSheet, View } from 'react-native'
import UIText from 'ui-elements/text'
import colors from 'styles/colors'

const UIBaloon: (args: any) => React$Node = ({
  children,
  style,
  type,
  ...otherProps
}) => {
  const getColorScheme = () => {
    switch (type) {
      case 'info':
        return {
          container: styles.uiBaloonInfo,
          text: styles.uiBaloonNormalInfo,
        }
      default:
        return {
          container: styles.uiBaloonNormal,
          text: styles.uiBaloonNormalText,
        }
    }
  }

  const baloonStyle = getColorScheme()

  return (
    <View style={[styles.uiBaloon, baloonStyle.container, style]}>
      {typeof children === 'string' ? (
        <UIText style={baloonStyle.text} {...otherProps}>
          {children}
        </UIText>
      ) : (
        children
      )}
    </View>
  )
}

const styles = StyleSheet.create({
  uiBaloon: {
    padding: 12,
    borderRadius: 12,
    borderWidth: StyleSheet.hairlineWidth,
  },
  uiBaloonInfo: {
    borderColor: colors.lightBlue,
    backgroundColor: colors.lighterBlue,
  },
  uiBaloonNormalInfo: {
    color: colors.black,
  },
  uiBaloonNormal: {
    borderColor: colors.darkGray,
    backgroundColor: colors.lightGray,
  },
  uiBaloonNormalText: {
    color: colors.black,
  },
})

export default UIBaloon
