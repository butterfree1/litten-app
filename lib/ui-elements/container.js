/**
 * @format
 * @flow
 */

import { StyleSheet, View } from 'react-native'

const UIContainer: (args: any) => React$Node = ({
  children,
  style,
  ...otherProps
}) => (
  <View {...otherProps} style={StyleSheet.compose(styles.uiContainer, style)}>
    {children}
  </View>
)

const styles = StyleSheet.create({
  uiContainer: {
    width: '75%',
  },
})

export default UIContainer
