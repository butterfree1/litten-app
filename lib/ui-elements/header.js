/**
 * @format
 * @flow
 */

import { StyleSheet, Text } from 'react-native'
import colors from 'styles/colors'

const UIHeader: (args: any) => React$Node = ({
  children,
  style,
  subheader = false,
  thin = false,
  ...otherProps
}) =>
  children ? (
    <Text
      {...otherProps}
      style={[
        StyleSheet.compose(styles.uiHeader, style),
        subheader ? styles.uiSubHeader : undefined,
        thin ? styles.uiHeaderThin : undefined,
      ]}>
      {children}
    </Text>
  ) : null

const styles = StyleSheet.create({
  uiHeader: {
    fontSize: 24,
    fontWeight: '700',
    color: colors.black,
    paddingTop: 2,
    paddingBottom: 2,
    marginTop: 2,
    marginBottom: 2,
  },
  uiHeaderThin: {
    fontWeight: '200',
  },
  uiSubHeader: {
    fontSize: 16,
  },
})

export default UIHeader
