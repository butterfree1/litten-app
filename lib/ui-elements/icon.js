/**
 * @format
 * @flow
 */

import { StyleSheet, View } from 'react-native'
import UIImage from 'ui-elements/image'
import colors from 'styles/colors'
import {
  elevated as elevatedStyle,
  veryElevated as veryElevatedStyle,
} from 'styles/common'
import {
  UI_ICON_SIZE_MEDIUM,
  UI_ICON_SIZE_SMALL,
  UI_ICON_SIZE_MINI,
} from 'utils/constants'

const UIText: (args: any) => React$Node = ({
  circle = false,
  icon,
  IconComponent,
  iconStyle,
  selected = false,
  size = 'small',
  style,
  ...otherProps
}) => {
  const getSizeStyle = () => {
    switch (size) {
      case 'medium':
        return {
          circle: styles.uiIconContainerCircleMedium,
          dimensions: UI_ICON_SIZE_MEDIUM,
          elevation: elevatedStyle,
          main: styles.uiIconContainerMainMedium,
        }
      case 'mini':
        return {
          circle: styles.uiIconContainerCircleMini,
          dimensions: UI_ICON_SIZE_MINI,
          elevation: elevatedStyle,
          main: styles.uiIconContainerMainMini,
        }
      default:
        return {
          circle: styles.uiIconContainerCircleSmall,
          dimensions: UI_ICON_SIZE_SMALL,
          elevation: veryElevatedStyle,
          main: styles.uiIconContainerMainSmall,
        }
    }
  }

  const sizeStyle = getSizeStyle()
  const elevationStyle = sizeStyle.elevation
  const mainSelected = selected ? styles.uiIconContainerMainSelected : undefined
  const iconSelected = selected ? styles.uiIconSelected : undefined
  const fillColor =
    iconStyle?.tintColor ?? (selected ? colors.white : colors.blue)

  return (
    <View
      {...otherProps}
      style={[
        styles.uiIconContainer,
        sizeStyle.main,
        circle ? sizeStyle.circle : undefined,
        elevationStyle,
        mainSelected,
        style,
      ]}>
      {icon && !IconComponent && (
        <UIImage
          source={icon}
          style={[styles.uiIcon, iconSelected, iconStyle]}
        />
      )}
      {!icon && IconComponent && (
        <IconComponent
          height={sizeStyle.dimensions / 2}
          width={sizeStyle.dimensions / 2}
          fill={fillColor}
        />
      )}
    </View>
  )
}

const styles = StyleSheet.create({
  uiIconContainer: {
    aspectRatio: 1,
    borderRadius: 4,
    backgroundColor: colors.white,
  },
  uiIconContainerMainSelected: {
    backgroundColor: colors.blue,
  },
  uiIconContainerMainMedium: {
    height: UI_ICON_SIZE_MEDIUM,
    padding: UI_ICON_SIZE_MEDIUM / 4,
  },
  uiIconContainerCircleMedium: {
    borderRadius: UI_ICON_SIZE_MEDIUM / 2,
  },
  uiIconContainerMainSmall: {
    height: UI_ICON_SIZE_SMALL,
    padding: UI_ICON_SIZE_SMALL / 4,
  },
  uiIconContainerCircleSmall: {
    borderRadius: UI_ICON_SIZE_SMALL / 2,
  },
  uiIconContainerMainMini: {
    height: UI_ICON_SIZE_MINI,
    padding: UI_ICON_SIZE_MINI / 4,
  },
  uiIconContainerCircleMini: {
    borderRadius: UI_ICON_SIZE_MINI / 2,
  },
  uiIcon: {
    height: '100%',
    width: '100%',
    tintColor: colors.blue,
  },
  uiIconSelected: {
    tintColor: colors.white,
  },
})

export default UIText
