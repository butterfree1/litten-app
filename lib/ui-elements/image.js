/**
 * @format
 * @flow
 */

import { Image } from 'react-native'

const UIImage: (args: any) => React$Node = ({
  noCache = false,
  resizeMode = 'contain',
  source,
  ...otherProps
}) => {
  const getSource = () => {
    if (!noCache && typeof source === 'object' && source.uri) {
      return {
        ...source,
        cache: 'force-cache',
      }
    }
    return source
  }

  return <Image source={getSource()} resizeMode={resizeMode} {...otherProps} />
}

export default UIImage
