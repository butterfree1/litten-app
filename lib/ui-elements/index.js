import UIAvatar from './avatar'
import UIBadge from './badge'
import UIBaloon from './baloon'
import UIButton from './button'
import UIContainer from './container'
import UIDropdown from './dropdown'
import UIHeader from './header'
import UIIcon from './icon'
import UIImage from './image'
import UIImagePlaceholder from './image-placeholder'
import UIInput from './input'
import UILink from './link'
import UIListItem from './list-item'
import UILoader from './loader'
import UIMessage from './message'
import UIModal from './modal'
import UIPasswordInput from './password-input'
import UIProgress from './progress'
import UIPrompt from './prompt'
import UISearch from './search'
import UISelect from './select'
import UISeparator from './separator'
import UISlider from './slider'
import UISwitch from './switch'
import UIText from './text'
import UITextArea from './textarea'
import UITooltip from './tooltip'

export {
  UIAvatar,
  UIBadge,
  UIBaloon,
  UIButton,
  UIContainer,
  UIDropdown,
  UIHeader,
  UIIcon,
  UIImage,
  UIImagePlaceholder,
  UIInput,
  UILink,
  UIListItem,
  UILoader,
  UIMessage,
  UIModal,
  UIPasswordInput,
  UIProgress,
  UIPrompt,
  UISearch,
  UISelect,
  UISeparator,
  UISlider,
  UISwitch,
  UIText,
  UITextArea,
  UITooltip,
}
