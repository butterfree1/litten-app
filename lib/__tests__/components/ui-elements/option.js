/**
 * @format
 */

import { UIOption } from 'ui-elements'
import TestRenderer from 'react-test-renderer'

describe('Snapshot test for the "UISwitch" component', () => {
  it('renders correctly', () => {
    const items = [
      { key: 'one', label: 'One', value: 'one' },
      { key: 'two', label: 'Two', value: 'two' },
    ]
    const selectedValue = 'one'
    const element = TestRenderer.create(
      <UIOption
        items={items}
        label="Some text here"
        selectedValue={selectedValue}
      />,
    ).toJSON()
    expect(element).toMatchSnapshot()
  })
})
