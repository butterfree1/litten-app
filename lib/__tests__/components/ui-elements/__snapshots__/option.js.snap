// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for the "UISwitch" component renders correctly with description 1`] = `
Array [
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#c4c4c4",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 12,
        "paddingBottom": 12,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 12,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#000000",
          "flex": 1,
          "fontSize": 16,
          "fontWeight": "400",
        }
      }
    >
      Some text here
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "200",
          },
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      One
    </Text>
  </View>,
  <Text
    style={
      Array [
        Array [
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "200",
          },
          Object {
            "marginTop": 8,
          },
        ],
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    Some more text here
  </Text>,
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    supportedOrientations={
      Array [
        "portrait",
      ]
    }
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#d0d4da",
          "height": 215,
          "justifyContent": "center",
        }
      }
    >
      <View>
        <RNCPicker
          items={
            Array [
              Object {
                "label": "One",
                "testID": undefined,
                "textColor": undefined,
                "value": "one",
              },
              Object {
                "label": "Two",
                "testID": undefined,
                "textColor": undefined,
                "value": "two",
              },
            ]
          }
          onChange={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </Modal>,
]
`;

exports[`Snapshot test for the "UISwitch" component renders correctly without description 1`] = `
Array [
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#c4c4c4",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 12,
        "paddingBottom": 12,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 12,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#000000",
          "flex": 1,
          "fontSize": 16,
          "fontWeight": "400",
        }
      }
    >
      Some text here
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "200",
          },
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      One
    </Text>
  </View>,
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    supportedOrientations={
      Array [
        "portrait",
      ]
    }
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#d0d4da",
          "height": 215,
          "justifyContent": "center",
        }
      }
    >
      <View>
        <RNCPicker
          items={
            Array [
              Object {
                "label": "One",
                "testID": undefined,
                "textColor": undefined,
                "value": "one",
              },
              Object {
                "label": "Two",
                "testID": undefined,
                "textColor": undefined,
                "value": "two",
              },
            ]
          }
          onChange={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </Modal>,
]
`;
